packages/@webex/plugin-meetings/src/recording-controller/index.ts

```diff
@@ -0,0 +1,214 @@
+import PermissionError from '../common/errors/permission';
+import {CONTROLS, HTTP_VERBS} from '../constants';
+import LocusInfo from '../locus-info';
+import MeetingRequest from '../meeting/request';
+import MeetingUtil from '../meeting/util';
+import RecordingAction from './enums';
+import Util from './util';
+import LoggerProxy from '../common/logs/logger-proxy';
+
+/**
+ * @description Recording manages the recording functionality of the meeting object, there should only be one instantation of recording per meeting
+ * @export
+ * @private
+ * @class Recording
+ */
+export default class RecordingController {
+  /**
+   * @instance
+   * @type {MeetingRequest}
+   * @private
+   * @memberof RecordingController
+   */
+  private request: MeetingRequest;
+  /**
+   * @instance
+   * @type {MeetingRequest}
+   * @private
+   * @memberof RecordingController
+   */
+  private info: LocusInfo;
+
+  /**
+   * @instance
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private serviceUrl: string;
+
+  /**
+   * @instance
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private sessionId: string;
+
+  /**
+   * @instance
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private locusUrl: string;
+
+  /**
+   * @instance
+   * @type {string}
+   * @private
+   * @memberof RecordingInfo
+   */
+  private locusId: string;
+
+  /**
+   * @param {MeetingRequest} request
+   * @param {LocusInfo} info
+   * @constructor
+   * @memberof RecordingController
+   */
+  constructor(request: MeetingRequest, info: LocusInfo) {
+    this.initialize(request, info);
+  }
+
+  /**
+   * @param {MeetingRequest} request
+   * @param {LocusInfo} info
+   * @returns {void}
+   * @private
+   * @memberof RecordingController
+   */
+  private initialize(request: MeetingRequest, info: LocusInfo) {
```

Comment:
why do we need to save the LocusInfo here ? can we just update the specific info rather then the whole locusInfo object !?

Reply:
Yeah we could do it that way as well, or the recording class can parse the locusInfo, it's sort of 6 and one half dozen in this case because it's an internal implementation and is not exposed as an API

Reply:
I also think it's better not to save the whole info, it will save a bit of memory (locus info can grow quite large, especially for big meetings) and it's also confusing when debugging things and seeing same locus info duplicated in many places.
Anyway, you're already extracting the bits you need in the extract() method, so why save the whole info?

Reply:
Cool. I'm on the same page here and I can update the PR for this